main.go

```diff
@@ -563,3 +568,56 @@ func deployCloudComponents(plan types.Plan) error {
 
 	return nil
 }
+
+func featureEnabled(features []string, secretFeatures []string) bool {
+	for _, feature := range features {
+		for _, secretFeature := range secretFeatures {
+			if feature == secretFeature {
+				return true
+			}
+		}
+	}
+	return false
+}
+
+func filterFeatures(plan types.Plan) types.Plan {
+	plan.Features = append(plan.Features, types.DefaultFeature)
+
+	if (plan.S3 != types.S3{}) {
+		plan.Features = append(plan.Features, types.S3Bucket)
+	}
+
+	if (plan.Github != types.Github{}) {
+		plan.Features = append(plan.Features, types.GitHub)
+	}
+
+	if plan.TLS == true {
+		plan = filterDNSFeature(plan)
+	}
+
+	if plan.EnableOAuth == true {
+		plan.Features = append(plan.Features, types.Auth)
+	}
+
+	if plan.BasicAuth == true {
+		plan.Features = append(plan.Features, types.BasicAuth)
+	}
+
+	if plan.Registry != "" {
+		plan.Features = append(plan.Features, types.Registry)
+	}
+	return plan
+}
+
+func filterDNSFeature(plan types.Plan) types.Plan {
```

Comment:
Should should have a test, it should be relatively simple to add. 

Where is the default case for where none is recognised? We'll need to return `(types.Plan, error)` so we can inform the user the feature was requested, but no secret was given for the DNS.

Reply:
Good proposal ^ I will add that

Reply:
Added the error and also tests ✔️